### PR TITLE
[DOC] Replace the word 'defaulted' with better/correct alternative.

### PR DESCRIPTION
**Pull request by smehringer**
`defaulted` as an adjective means something like `fails to fulfil an obligation`. 
We sometimes do not use this word correctly (since we always mean `set to default`).

Only `to be defaulted` actually means `set to default`. To avoid this possible confusing, some correct sentences were also changed.

### CODE REVIEW

File: include/seqan3/alphabet/composite/alphabet_variant.hpp [hunk 1]
```diff
@@ -268,7 +268,7 @@ class alphabet_variant : public alphabet_base<alphabet_variant<alternative_types
      *
      * \details
      *
-     * Most assignments happen through implicit conversion and the defaulted assignment operator. This is for the rest.
+     * Most assignments happen through implicit conversion and the default assignment operator. This is for the rest.
```
Comment (Irallia) [inline include/seqan3/alphabet/composite/alphabet_variant.hpp:271]:
```suggestion
     * Most assignments happen through implicit conversion and the default assignment operator. This alphabet_variant is for the rest.
```

**Review by Irallia — APPROVED** [review]
LGFM, just one word I would add, for a bit more meaning 💅

File: include/seqan3/io/detail/misc_input.hpp [hunk 2]
```diff
@@ -71,7 +71,7 @@ inline bool starts_with(ref_t && reference, query_t && query)
 /*!\brief Depending on the magic bytes of the given stream, return a decompression stream or forward the primary stream.
  * \param[in] primary_stream The primary (device) stream for reading.
  * \param[in,out] filename  The associated filename; compression extensions will be stripped. [optional]
- * \returns A pointer to the secondary stream with defaulted or NOP'ed deleter.
+ * \returns A pointer to the secondary stream with a default deleter or a NOP'ed deleter.
```
Comment (rrahn) [inline include/seqan3/io/detail/misc_input.hpp:74]:
```suggestion
 * \returns A pointer to the secondary stream with a default deleter or a nop-deleter.
```

File: include/seqan3/io/detail/misc_output.hpp [hunk 3]
```diff
@@ -32,7 +32,7 @@ namespace seqan3::detail
 /*!\brief Depending on the given filename/extension, create a compression stream or just forward the primary stream.
  * \param[in] primary_stream The primary (uncompressed) stream for writing.
  * \param[in,out] filename  The associated filename; compression extensions will be stripped.
- * \returns A pointer to the secondary stream with defaulted or NOP'ed deleter.
+ * \returns A pointer to the secondary stream with a default deleter or a NOP'ed deleter.
```
Comment (rrahn) [inline include/seqan3/io/detail/misc_output.hpp:35]:
```suggestion
 * \returns A pointer to the secondary stream with a default deleter or a nop-deleter.
```

**Review by rrahn — APPROVED** [review]

**Pull request merged**